### PR TITLE
Home page

### PR DESCRIPTION
**Pull request by Saroko-dnd**
I created home page for test react application with news. I will add more css  later if it will be necessary.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import 'bootstrap/scss/bootstrap.scss';
+
+import './assets/scss/index.scss';
+import App from './app/App';
+
+render(
+    <BrowserRouter>
+        <App />
+    </BrowserRouter>,
+    document.getElementById('reactApp'),
```
Comment (abramenal) [inline src/index.js]: I think DOM values (like element ID here) shouldn't be in camelCase.

File: src/assets/scss/_reset.scss [hunk 2]
```diff
@@ -0,0 +1,139 @@
+html,
```
Comment (abramenal) [inline src/assets/scss/_reset.scss]: I think it does make sense to use any package instead of copying the reset to every your new project.

Comment (abramenal) [reply src/assets/scss/_reset.scss]: You can also try to practice creating your own package if you wish to leave this content as is

File: src/app/applicationComponents/Header/index.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                exact
+                // activeStyle={{ color: 'white' }}
+            >
+                Ultra news
+            </NavLink>
+            <NavLink
+                className="nav-item nav-link"
+                to="/second"
+                exact
+                // activeStyle={{ color: 'white' }}
```
Comment (abramenal) [inline src/app/applicationComponents/Header/index.js]: Please scan through your files and remove unnecessary comments.

File: src/app/applicationComponents/Header/index.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+const Header = () => (
+    <header className="site-header d-flex align-items-center flex-column">
+        <header className="d-flex align-items-center">
+            <span>
+                <NewsIcon height="150" width="150" />
+            </span>
+            News from all over the world
+        </header>
+        <nav className="site-header-navigation nav nav-pills d-flex justify-content-around">
```
Comment (abramenal) [inline src/app/applicationComponents/Header/index.js]: I think the app's navigation is a separate component, not a part of header

File: src/app/applicationComponents/Main/index.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import { Switch, Route } from 'react-router-dom';
+
+import NewsList from '../../reusableComponents/NewsList';
+import newsAppConfig from '../../configFiles/newsAppConfig';
+
+class Main extends React.Component {
```
Comment (abramenal) [inline src/app/applicationComponents/Main/index.js]: What is "main" component? for what does it stand for?
Please name it more specific to it's functionality

File: src/app/applicationComponents/Main/index.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+                <Route
+                    exact
+                    path="/"
+                    render={() => (
+                        <main
+                            className={`${
+                                this.props.className
+                            } container-fluid`}
+                        >
+                            <section className="row">
```
Comment (abramenal) [inline src/app/applicationComponents/Main/index.js]: I think this news list is a separate component too

File: src/app/reusableComponents/NewsList/index.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
```
Comment (abramenal) [inline src/app/reusableComponents/NewsList/index.js]: I cannot accept the existing folder and file structure, please check the official docs for further info

File: src/app/reusableComponents/svg/NewsIcon/index.js [hunk 8]
```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+
+import './NewsIcon.scss';
+
+const NewsIcon = props => (
+    <svg
```
Comment (abramenal) [inline src/app/reusableComponents/svg/NewsIcon/index.js]: If it is copied from somewhere I suggest to use a package instead

**Review by abramenal — CHANGES_REQUESTED** [review]

File: src/assets/scss/_reset.scss (hunk 2 above)
Comment (Saroko-dnd) [reply src/assets/scss/_reset.scss]: Done. I have created new package  scss-reset-by-igor-saroko for reset.scss file.

File: src/index.js (hunk 1 above)
Comment (Saroko-dnd) [reply src/index.js]: Done. camelCase was removed from  element with id reactApp. Now it has id react-app.

File: src/app/applicationComponents/Header/index.js (hunk 3 above)
Comment (Saroko-dnd) [reply src/app/applicationComponents/Header/index.js]: Done.

File: src/app/applicationComponents/Header/index.js (hunk 4 above)
Comment (Saroko-dnd) [reply src/app/applicationComponents/Header/index.js]: Done. 
I have created NavigationPanel component.

File: src/app/applicationComponents/Main/index.js (hunk 5 above)
Comment (Saroko-dnd) [reply src/app/applicationComponents/Main/index.js]: Renamed to RouteContentContainer.

File: src/app/applicationComponents/Main/index.js (hunk 6 above)
Comment (Saroko-dnd) [reply src/app/applicationComponents/Main/index.js]: Done.
Created new component - NewsColumns.

File: src/app/reusableComponents/NewsList/index.js (hunk 7 above)
Comment (Saroko-dnd) [reply src/app/reusableComponents/NewsList/index.js]: Done.
Application is divided into components, containers, constants, and utilities.

File: src/app/components/ArticlesList/index.jsx [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+import Article from '../../components/Article';
+
+import createUuidv4 from '../../utils/createUuidv4';
+
+import './scss/index.scss';
+
+const ArticlesList = (props) => {
+    const arrayOfNews = props.articles.map(article => (
+        <Article
+            key={createUuidv4()}
```
Comment (abramenal) [inline src/app/components/ArticlesList/index.jsx]: The key shouldn't be generated randomly, please check it's mission in scope of reconcilliation.

File: src/app/components/ArticlesList/index.jsx [hunk 10]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Article from '../../components/Article';
+
+import createUuidv4 from '../../utils/createUuidv4';
+
+import './scss/index.scss';
+
+const ArticlesList = (props) => {
```
Comment (abramenal) [inline src/app/components/ArticlesList/index.jsx]: Please apply props destructuring for each component.

**Review by abramenal — CHANGES_REQUESTED** [review]

File: src/app/reusableComponents/svg/NewsIcon/index.js (hunk 8 above)
Comment (Saroko-dnd) [reply src/app/reusableComponents/svg/NewsIcon/index.js]: Done.
I created new npm package for my svg icons written as React components.

File: src/app/components/ArticlesList/index.jsx (hunk 9 above)
Comment (Saroko-dnd) [reply src/app/components/ArticlesList/index.jsx]: Done.
All keys have been changed to news url.

File: src/app/components/ArticlesList/index.jsx (hunk 10 above)
Comment (Saroko-dnd) [reply src/app/components/ArticlesList/index.jsx]: Done.

**Review by abramenal — APPROVED** [review]

**Pull request merged**